apps/server/src/modules/task/uc/task-copy.uc.spec.ts

```diff
@@ -270,7 +265,8 @@ describe('task copy uc', () => {
 					const task = taskFactory.buildWithId();
 					userRepo.findById.mockResolvedValue(user);
 					taskRepo.findById.mockResolvedValue(task);
-					authorisation.hasPermission.mockImplementation((u: User, e: AuthorizableObject) => e !== task);
+					// authorisation should not be mocked
```

Comment:
Why?

Reply:
We have only 2 files in our application that mock the service in this way. In both it is not needed. The service is syncron and can be used directly in test. The uc must also validate that the execution they want to trigger match to a knowen rule in the knowen context. 
The authorisation service can not validate this. With the decission that api test should not validate the authorisation variantions, this is the last possible place that can do this.

Reply:
Is note as arc weekly point.

Reply:
The rule it self must be tested. But we have not test that validate that the request context that passed to the UC give the expected result for the authorisation. The authorisation service do not know the rule later. The rule and the authorisation service do not know if the params in UC correctly created and set.
The UC it self do not know if any rule match and work like expected.

 In this case we lost the test for the "real" connection of the test. Or let say we have a tested A and tested B, but we have no test that validate if A<->B really work.